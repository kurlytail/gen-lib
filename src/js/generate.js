function generate() {}

export default generate;
