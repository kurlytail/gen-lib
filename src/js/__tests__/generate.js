import FS from 'fs';
import mkdirp from 'mkdirp';

import { getOverwriteOption, manageFileNames, generateFileData } from '../generate';

jest.mock('fs');
jest.mock('mkdirp');
jest.mock('../logger');

const FIXTURES = {
    OPTION_OUTPUT: { output: 'some/directory' },
    TEMPLATE_DESCRIPTION_NOOVERWRITE: { template: '/test' },
    FILENAMES: '/file',
    EMPTY_GENERATOR: {
        options: {},
        design: {},
        map: {},
        extensionBuilder: {}
    }
};

describe('# generate', () => {
    describe('## manageFileNames', () => {
        beforeEach(() => {
            mkdirp.mockReset();
        });
        it('### should return filenames for filename with no options', () => {
            let fileNames = manageFileNames({}, 'test');
            expect(fileNames).toMatchSnapshot();
            expect(mkdirp.sync).toBeCalled();
        });
        it('### should return filenames for filename with output options', () => {
            let fileNames = manageFileNames(FIXTURES.OPTION_OUTPUT, 'test');
            expect(fileNames).toMatchSnapshot();
            expect(mkdirp.sync).toBeCalled();
        });
        it('### should return filenames for absolute filename with output options', () => {
            let fileNames = manageFileNames(FIXTURES.OPTION_OUTPUT, '/test');
            expect(fileNames).toMatchSnapshot();
            expect(mkdirp.sync).toBeCalled();
        });
    });

    describe('## generateFileData', () => {
        beforeEach(() => {
            FS.readFileSync.mockReset();
            FS.existsSync.mockReset();
        });
        it('### should generate template file', () => {
            FS.readFileSync.mockReturnValue('some template');
            const output = generateFileData(
                FIXTURES.EMPTY_GENERATOR,
                FIXTURES.TEMPLATE_DESCRIPTION_NOOVERWRITE,
                FIXTURES.FILENAMES
            );
            expect(output).toMatchSnapshot();
        });
    });
});
