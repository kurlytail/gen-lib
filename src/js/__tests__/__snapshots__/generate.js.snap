// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# generate ## generateFileData ### should generate template file 1`] = `"some template"`;

exports[`# generate ## manageFileNames ### should return filenames for absolute filename with output options 1`] = `"/test"`;

exports[`# generate ## manageFileNames ### should return filenames for filename with no options 1`] = `"test"`;

exports[`# generate ## manageFileNames ### should return filenames for filename with output options 1`] = `"some/directory/test"`;
